File: LogisimPlus/LogisimPlus/logisim/gui/generic/CardPanel.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// To produce customized conversions, purchase a Premium Edition license:
// https://www.tangiblesoftwaresolutions.com/product-details/java-to-csharp-converter.html
// ====================================================================================================

using System.Collections.Generic;

/* Copyright (c) 2010, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

namespace logisim.gui.generic
{


	public class CardPanel : JPanel
	{
		private List<ChangeListener> listeners;
		private string current;

		public CardPanel() : base(new CardLayout())
		{
			listeners = new List<ChangeListener>();
			current = "";
		}

		public virtual void addChangeListener(ChangeListener listener)
		{
			listeners.Add(listener);
		}

		public virtual void addView(string name, Component comp)
		{
			add(comp, name);
		}

		public virtual string View
		{
			get
			{
				return current;
			}
			set
			{
				if (string.ReferenceEquals(value, null))
				{
					value = "";
				}
				string oldChoice = current;
				if (!oldChoice.Equals(value))
				{
					current = value;
					((CardLayout) getLayout()).show(this, value);
					ChangeEvent e = new ChangeEvent(this);
					foreach (ChangeListener listener in listeners)
					{
						listener.stateChanged(e);
					}
				}
			}
		}


	}

}

File: logisim/app/src/main/java/logisim/gui/generic/CardPanel.java
/* Copyright (c) 2010, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

package logisim.gui.generic;

import java.awt.CardLayout;
import java.awt.Component;
import java.util.ArrayList;

import javax.swing.JPanel;
import javax.swing.event.ChangeEvent;
import javax.swing.event.ChangeListener;

public class CardPanel extends JPanel {
	private ArrayList<ChangeListener> listeners;
	private String current;

	public CardPanel() {
		super(new CardLayout());
		listeners = new ArrayList<>();
		current = "";
	}

	public void addChangeListener(ChangeListener listener) {
		listeners.add(listener);
	}

	public void addView(String name, Component comp) {
		add(comp, name);
	}

	public String getView() {
		return current;
	}

	public void setView(String choice) {
		if (choice == null)
			choice = "";
		String oldChoice = current;
		if (!oldChoice.equals(choice)) {
			current = choice;
			((CardLayout) getLayout()).show(this, choice);
			ChangeEvent e = new ChangeEvent(this);
			for (ChangeListener listener : listeners) listener.stateChanged(e);
		}
	}

}
